### PR TITLE
Adapt PR template to issue format

### PR DESCRIPTION
**Pull request by ferhoyos**
**Describe the change**

Adapt PR template to recently changed issue template and move it to the same folder (`.github`).

New sections added to complete PR description

### CODE REVIEW

File: .github/pull_request_template.md
Comment (nrfox) [inline .github/pull_request_template.md:11]: While including tests should be a requirement and it's the PR author's responsibility to include tests and it's the PR reviewer's responsibility to ensure tests are included, I'm not sure we should make folks write out what/where the tests are. Maybe a general PR checklist includes relevant tests. Something like:
- [ ] This PR includes tests (unit or E2E).

File: .github/pull_request_template.md
Comment (nrfox) [inline .github/pull_request_template.md:1]: :+1: I like the headers better

File: .github/pull_request_template.md
Comment (ferhoyos) [reply .github/pull_request_template.md]: I have added PR checklist section according to your suggestion. What any other item do you think it could be added to PR checklist? In my case I always forget to test in Firefox and the browser checklist would help me to remember. For frontend we have eslint rules so no need to add checklist item, but maybe in backend it would be good to add some `Review types` or similar action.

Comment (jshaughn) [reply .github/pull_request_template.md]: Is it really worth asking people to test on multiple browsers?  I know there are occasional issues, like FF scrollbar in logs, but in general browser incompatibility is rare.

Comment (ferhoyos) [reply .github/pull_request_template.md]: Since browser incompatibility is just related to CSS styles, I think a visual quick lookup in the other browser supported does not take too much time.

File: .github/pull_request_template.md
Comment (josunect) [inline .github/pull_request_template.md:5]: I would add something more clear like "Steps to test the PR"

File: .github/pull_request_template.md
Comment (josunect) [inline .github/pull_request_template.md:9]: I think this is something that we do with labels? I would add as less information as possible, so it is less probable to omit them.

File: .github/pull_request_template.md
Comment (josunect) [inline .github/pull_request_template.md:20]: We can add the keywords Fixes or Closes

File: .github/pull_request_template.md
Comment (ferhoyos) [reply .github/pull_request_template.md:9]: Problems with labels is that the developer proactively has to add the label. For me PR checklist is to remind the developer some things that can be forgotten (like a TODO list).

File: .github/pull_request_template.md
Comment (ferhoyos) [reply .github/pull_request_template.md]: I have added example with Closes keyword

**Comment by ferhoyos** [comment]
Due to feedback received I have decided to remove PR checklist (most of the stuff to be checked is done by linters). Maybe it is better to keep the PR template as simple as possible.

**Comment by ScriptingShrimp** [comment]
Hello @jshaughn , team
with respect to recent discussion on Service Mesh Sync regarding the backlog size (especially for testing) I cannot approve the template changes as its proposed in this PR. It simply does not contain anything about automation and delivering test coverage, therefore I conclude we have no commitment to reduce automation backlog. 

Adding tests to pull requests is a crucial step in ensuring code quality, reliability, and maintainability. Here are some best practices for adding tests to pull requests:

1. Write Unit Tests
2. Use Descriptive Test Names 
3. Cover Edge Cases (in case the PR is bugfix or similar) 
4. Keep Tests DRY (Don't Repeat Yourself, do not introduce technical debt)
5. Continuous Integration (CI should be adjusted to work with new test requirements etc)
6. Document Testing Strategy (last but the most important - there must be settled strategy at least on Epic level)

On the other hand I can add few examples how we increase backlog size and testing debt. You might find some more reasonable than others, but we should conclude what actions to take to narrow down this problematic. 

1. No testing strategy established during planning or early phase of development (no requirement set, but should be obligation in each Epic or Task)
2. Changes are too complex or the PR is too big, so covering the changes is really expensive (complex tasks)
3. Feedback on the proposed changes before investing time in writing tests (adjust expectations, see point 1, usually ends up as 2)
4. Integration with external services or dependencies that are difficult to replicate
5. Temporary Workarounds or Legacy Code Constraints
6. Lack of morale or discipline to contribute tests

Looking for team feedback,
Thank You

**Comment by jshaughn** [comment]
>  It simply does not contain anything about automation and delivering test coverage, therefore I conclude we have no commitment to reduce automation backlog.

Well, I don't agree that we have no commitment to reduce automation backlog.  We have entire epics in progress for adding automation, see https://github.com/kiali/kiali/issues/6361.  And we are approaching 10,000 lines under the `cypress` project folder.  That is a 33% increase over v1.65 and 25% increase over v1.73.  And that does not include additions in unit tests, molecule tests, etc.  This doesn't mean we don't have a backlog of automation work, we do, and we probably always will.  We also have a large backlog of feature work, and probably always will.  Engineering is committed to both, it's a balancing act.

Having said that, I do think we could add a header to the PR like:

`### Automation testing added?`

If nothing else it would be a reminder.  It would also allow a contributor to say, "No, not needed because ....".

File: .github/pull_request_template.md
Comment (jshaughn) [reply .github/pull_request_template.md:11]: > if related to a **GitHub** issue, verify you mention and close the Issue (e.g., Closes #100)
I'm -1 on adding anything related to our downstream RH processes to upstream Jira PR templates.  If we happen to be working on something related to downstream a reference is typically added, without being prompted.

Comment (ScriptingShrimp) [reply .github/pull_request_template.md:11]: Whatever suits our core contributors. Maybe we want to have short meeting to sort this out?

Comment (jshaughn) [reply .github/pull_request_template.md:11]: I think we can do this async, @ferhoyos, is this ready for another round of review?

File: .github/pull_request_template.md
Comment (ferhoyos) [inline .github/pull_request_template.md:4]:
```suggestion
### Automation testing 

If applicable, explain the case scencarios covered by **unit / integration / e2e** tests created for this PR.

```

With all the feedback received I think most of us agree to have some kind of testing section (I don't see a PR checklist just to check if unit/e2e tests have been created). Brief explanation of the test case scenarios can be useful to reviewer (and qa) to determine if the PR is well covered by tests or not.

File: .github/pull_request_template.md
Comment (ferhoyos) [inline .github/pull_request_template.md:11]:
```suggestion
If this PR is related to a github issue, please link it here (https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
```

I think it is better to refer to github link that shows how to link an issue in a PR. With the example you have the risk to link multiple PRs to issue 100 if the comment is not removed.

Comment (ferhoyos) [reply .github/pull_request_template.md:11]: I'm -1 to add downstream reference in PR template. Maybe for some specific PR you can add JIRA link, but I think this should not be the default option in an upstream project.

**Review by ferhoyos — COMMENTED** [review]
I have added two suggestions with feedback received. I think we are ready for another round review @jshaughn.

File: .github/pull_request_template.md
Comment (jshaughn) [inline .github/pull_request_template.md:7]:
```suggestion
Recommendations for how to test this PR.  Reminder that each PR should also include a "Test" label.
```

Non-blocking suggestion.  We could add to our process that each PR must add one of the test labels.

**Review by jshaughn — DISMISSED** [review]
This is fine for me, I did add one non-blocking suggestion.

**Review by jshaughn — APPROVED** [review]
re-approving after update

**Comment by ferhoyos** [comment]
@ScriptingShrimp are you ok with the final proposal? @jshaughn I think that you have to merge this PR as admin since github action is not executed.

File: .github/pull_request_template.md
Comment (ScriptingShrimp) [reply .github/pull_request_template.md:4]: >  Brief explanation of the test case scenarios
+1

About testing strategy: we usually don't have it, or late in the development cycle. Is there such a thing like template for epic, or epic checklist?

Comment (ferhoyos) [reply .github/pull_request_template.md:4]: Currently there is no such epic template. Since there are very few epics I am not sure we should have a template just for epics. If you see testing strategy is missing in one of the epics, just contact the epic owner to discuss about it.

**Comment by jshaughn** [comment]
@ferhoyos Will merge if we get approval from @ScriptingShrimp .

**Pull request merged**
